src/internal/kopia/upload.go

```diff
@@ -235,38 +235,229 @@ func (cp *corsoProgress) get(k string) *itemDetails {
 	return cp.pending[k]
 }
 
-func collectionEntries(
-	ctx context.Context,
-	ctr func(context.Context, fs.Entry) error,
-	streamedEnts data.BackupCollection,
-	progress *corsoProgress,
-) (map[string]struct{}, error) {
-	if streamedEnts == nil {
-		return nil, nil
+// These define a small state machine as to which source to return an entry from
+// next. Since these are in-memory only values we can use iota. Phases are
```

Comment:
Understanding that we _can_ use iota here with relative safety, I still think it's better to avoid it by standard (ie: add a linter that says no iotas).  We've hit bugs too many times for it to feel comfortable.